vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -291,24 +293,16 @@ class OnboardingViewModel @AssistedInject constructor(
         currentJob = null
 
         when (action) {
-            OnboardingAction.ResetHomeServerType -> {
-                setState {
-                    copy(
-                            serverType = ServerType.Unknown
-                    )
-                }
+            OnboardingAction.ResetHomeServerType        -> {
+                setState { copy(serverType = ServerType.Unknown) }
             }
-            OnboardingAction.ResetHomeServerUrl  -> {
+            OnboardingAction.ResetHomeServerUrl         -> {
                 viewModelScope.launch {
                     authenticationService.reset()
                     setState {
                         copy(
                                 isLoading = false,
-                                homeServerUrlFromUser = null,
-                                homeServerUrl = null,
-                                loginMode = LoginMode.Unknown,
-                                serverType = ServerType.Unknown,
```

Comment:
resetting the server url shouldn't also reset the `server type` as the `server type` is actually the user selection from the "Matrix, EMS, Other" page